TesteLivro.java

```diff
@@ -0,0 +1,66 @@
+import java.io.IOException;
+import java.time.LocalDate;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+public class TesteLivro {
+    
+    private static Livros livros = new Livros();
+    
+    @Test
+	public void livroDuplicado() {
+    	Livro livro = new Livro("titulo", "resumo", "sumerio", (float) 30, 200, "isbn", LocalDate.parse("2020-12-03"), new Categoria("ux"));
+        livros.adiciona(livro);
+        try {
+        	Livro livro2 = new Livro("titulo", "resumo", "sumerio", (float) 30, 200, "isbn", LocalDate.parse("2020-12-03"), new Categoria("ux"));
```

Comment:
Aqui você precisa fazer cast pq um int não é um float mesmo. Se você quiser, da até para usar BigDecimal que é a classe mais utilizada para guardar valor e tal.

Reply:
Não entendo o motivo de usar BigDecimal pra um valor que é mais humano (dinheiros) do que feito pra máquina calcular. É por convenção mesmo?

Reply:
É uma ótima pergunta. A API de dinheiro -> https://javamoney.github.io/ ainda não está disponível :). Usamos o bigdecimal pq ele representa o valor decimal com mais precisão do que o float. Para ler mais, aqui pode ajudar => https://dzone.com/articles/never-use-float-and-double-for-monetary-calculatio Admito que não manjo dos detalhes da falta de precisão do double/float.